### DISCUSSION TITLE
Why are videos converted when I don't choose that option?

### DISCUSSION THREAD

**Discussion by gc435** [aclap-dev/video-downloadhelper#61] [Help and Q&A] [answered]
### Your question

When I download certain videos, they seem to be automatically converted.  I don't know why.  I **_don't_** choose **Download and Convert**.  I choose either **Quick Download** or **Download**.

1. Why are videos converted?
2. What's a good way to learn the source and destination formats?

I own a license.

### Addon & CoApp version and details

version: 8.1.4.0
beta: false
build date: 2024-01-04
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only first 16 fisrt characters): [redacted]
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.10",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\[redacted]",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "6.0"
}

**Comment by paulrouget** [comment]
They're not supposed to be converted. What makes you think they are?

[Maybe try version beta 8.2](https://downloadhelper.net/beta).

**Reply by gc435** [reply]
> They're not supposed to be converted. What makes you think they are?

I saw a QR code in the upper left corner of my downloaded videos.  I understand this to mean that the converter was used.  However, after a closer read of the docs, it seems a QR code is also displayed if aggregation comes into play.  Could this be it?

After seeing the QR code, I bought a license.

**Reply by GrampaWildWilly** [reply]
Well . . .  When VDH writes a QR code into something, that is because you don't have a license.  Now that you have a license, you are no longer seeing the QR code, right?

But the process of writing the QR code is not a conversion operation.  It is simply an aggregation operation without a conversion of the content from one type to another.  A conversion would be something like converting from mp4 to mkv, or converting from 1920x1080 to 1280x720, or other things of that nature.  Aggregating happens on certain types of content depending on the way the web site presents its content.  The most obvious example is YouTube.  VDH has no choice but to aggregate that content.  Actually, I should say that in the past tense.  VDH used to have to aggregate the content.  Since VDH started using ffmpeg as the download engine, that detail is handled within ffmpeg & there is no longer an explicit aggregation step.

It might clarify things if you posted a series of screen shots showing what has led you to your conclusion & your questions.

**Reply by gc435** [reply]
> Now that you have a license, you are no longer seeing the QR code, right?

Correct.  No QR code is displayed.

I probably misunderstood why the QR code was being displayed. I thought it was because the video was converted.  I now believe a QR code is displayed when video segments must be assembled into a final movie.  My source video is MPEG-DASH, so I believe that means a QR code would display in unlicensed versions of VDH.

**Reply by paulrouget** [reply]
We add a QR code on videos that are from youtube, and on videos that were download with the Download & Convert option.

Does that answer your question?
Any mismatch between this explanation and your experience?

**Reply by gc435** [reply]
My source video is from a site that appears to host the video on Vimeo.  This is all behind the scenes and I had to look at the HTML source to figure this out.  Maybe that's why the QR code was added.

**Reply by paulrouget** [reply]
I'm not sure to understand.

Is this problem solved or do you still have an issue?
Does my previous comment address your question?

**Reply by gc435** [reply]
I don't have a problem since I bought a license.  I'm now just trying to understand VDH better.  If you confirm that QR codes are added for downloads from Vimeo (or the Akamai CDN?), then that would explain everything.

**Reply by paulrouget** [reply, marked as answer]
Yes, we were re-encoding and adding a QR code.

**Comment by ibkhayyat** [comment]
I'm trying to download a video from YouTube, the video is being converted and a watermark added to all resolution options.

I have an active license. I tried from different browsers with no luck.

I would greatly appreciate any help.

**Reply by jcfrog** [reply]
What do you see in the license section of the settings?

**Reply by GrampaWildWilly** [reply]
Better yet, go into the VDH settings, click the `More settings` option, then click the button that says `Copy information to clipboard`.  Then paste whatever that button copied to your clipboard into a post here.

**Reply by ibkhayyat** [reply]
version: 9.0.2.12
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"mismatch":true,"key":"XXX","other_browser":"Chrome","this_browser":"Firefox"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "http_media_download_strategy": "inbrowser",
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\XXX\\dwhelper",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive"
        }
    },
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 2,
    "successfull_dl": 7
}

**Reply by GrampaWildWilly** [reply]
Ah.  There's the problem.  You have a license for VDH for Chrome but you're trying to use it with VDH for Firefox.  This is not allowed.  VDH for Chrome is one product.  VDH for Firefox is a different product.  They cost different amounts.  They have different features, mainly that Google does not permit VDH for Chrome to download from YouTube.  They can get away with that because Google owns the Chrome browser, the Chrome web store, & YouTube.  But Google can't impose such frivolous restrictions on Firefox because Firefox is independent of Google.  So VDH for Firefox is able to download from YouTube quite successfully.

There is a comment above that YouTube videos are being converted.  This is false.  YouTube videos are not converted.  They are aggregated.  That means that since YouTube presents the video & audio tracks separately, they must be put together, aggregated.  That's all the word aggregated means: put together.  Since you don't have a license for VDH for Firefox, VDH also adds a watermark during that aggregation step.  If you did have a proper license for VDH for Firefox, the aggregation would still occur but the watermark would not be written.  That's the primary purpose of a license: it skips the writing of the watermark.

Edit: I believe things are a bit different today.  VDH did not used to use ffmpeg to download content.  That includes from YouTube.  Used to be that VDH would download the video & audio tracks from YouTube, then execute an aggregation step to put the tracks together.  It would write the watermark at that time.  Ironically, the tool being used for this step was ffmpeg.  But now ffmpeg is the download engine.  Ffmpeg aggregates the video & audio tracks on the fly so the result file arrives as a complete media file with both video & audio tracks.  In order to add the watermark, VDH now re-encodes the file, as Paul said.  That is also a function performed by ffmpeg.  Are we confused enough yet?

Now, you must put the question to @jcfrog Jérôme in the `License & Purchase request` section of this forum what you want to do about your license.  You must decide if you want to keep your license for VDH for Chrome.  If you purchased it within the past 30 days, Jérôme will refund your money.  But you have to ask for it over there.  Alternatively, you may want to keep that license.  As a separate issue, you need to decide if you want to buy a license for VDH for Firefox.  You might want to have only the Firefox license & get a refund for your Chrome license.  Or you might want to keep both licenses.  Plenty of people have both licenses.  There's nothing blocking that possibility.  These are decisions you need to make.

**Reply by ibkhayyat** [reply]
Thank you very much for the detailed reply.
Well, I bought the license for Firefox, I've been using it on Firefox the whole time, and I've been able to download videos successfully without the watermark (for over 2 years). This is the first time I have had this issue.
As far as I remember, it mentioned it was a Chrome license, but it is the one to use on Firefox.
I'll let you know how things will go.

Thanks again, cheers

**Reply by GrampaWildWilly** [reply]
This is a line in the information you posted:

> license: {"mismatch":true,"key":"XXX","other_browser":"Chrome","this_browser":"Firefox"}

That says to me you're trying to use a Chrome license on Firefox.  If you say you have a Firefox license, there is something that needs to be sorted out.  When I collect that information, I get this:

> license: {"accepted":true,"email":"redacted","key":"redacted"}

(I manually inserted the word `redacted` because  I think all of that information should be kept private.)

Notice that your information says `"mismatch":true` while mine says `"accepted":true`.  That indicates to me that you've got something that needs to be fixed.  I suspect your best course of action is to post something in the `License & Purchase requests` area of this forum & get Jérôme to look into it.  He actually does work for VDH whereas I don't.  I'm just another user.  I can't help you with this sort of problem.

To be totally precise, the line from my information shows this:

`license: {"accepted":true,"email":"<removed>","key":"part of my key"}`

But GitHub treats anything within `< >` as a comment & doesn't display it.  I made it display by surrounding the line with the Code markers, something you can get by clicking this control above the text composition box:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/84dc87b0-af64-4c9a-a557-9abd23741e6d)

If you put &lt;whatever&gt; in regular text in a post here, it will magically become invisible.  So you have to take special measures to make the &lt;&gt; symbols appear in your post.  And what special measures did I take to make the symbols appear in this paragraph?  GitHub accepts the HTML character encodings `&lt;` & `&gt;`.

**Reply by ibkhayyat** [reply]
I checked my invoice again, it seems I ended up buying a Chrome license. I kept the Chrome one and bought a new one for Firefox.

Cheers
